### PR TITLE
Break out of infinite loop and bookmark on date window end

### PR DESCRIPTION
**Pull request by jacobrobertbaca**
# Description of change
Break out of infinite loop and bookmark on date window end

# Manual QA steps
 - ran a sync to make sure records were still coming through
 
# Risks
 - low
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tap_zendesk/streams.py [hunk 1]
```diff
@@ -203,13 +206,9 @@ def sync(self, state):
             # If we make it here, all quality checks have passed. Reset retry count.
             num_retries = 0
```
Comment (cosimon) [inline tap_zendesk/streams.py:207]: `num_retries` is always = 0, can we just remove it?

File: tap_zendesk/streams.py [hunk 2]
```diff
@@ -182,10 +182,13 @@ def sync(self, state):
...
-                end = start + datetime.timedelta(seconds=search_window_size)
-                LOGGER.info("users - Detected Search API response size too large. Cutting search window in half to %s seconds.", search_window_size)
-                continue
+                if search_window_size > 1:
+                    search_window_size = search_window_size // 2
+                    end = start + datetime.timedelta(seconds=search_window_size)
+                    LOGGER.info("users - Detected Search API response size too large. Cutting search window in half to %s seconds.", search_window_size)
+                    continue
+
+                raise Exception("users - Unable to get all users within minimum window of a single second ({}), found {} users within this timestamp. Zendesk can only provide a maximum of 1000 users per request. See: https://develop.zendesk.com/hc/en-us/articles/360022563994--BREAKING-New-Search-API-Result-Limits".format(parsed_start, users.count))
```
Comment (cosimon) [inline tap_zendesk/streams.py:191]: This Exception will conditionally execute based on the results of the `if search_window_size > 1:`, can we put it in an else block to call that out?

**Review by cosimon — CHANGES_REQUESTED** [review]

File: tap_zendesk/streams.py (hunk 1 above)
Comment (dmosorast) [reply tap_zendesk/streams.py:207]: The increment is hidden in this PR

File: tap_zendesk/streams.py (hunk 2 above)
Comment (dmosorast) [reply tap_zendesk/streams.py:191]: Pylint complains about that with some kind of "else after continue" :(

**Review by cosimon — APPROVED** [review]

**Pull request merged**